category_theory/limits/limits.lean

```diff
@@ -458,6 +462,12 @@ def colimit.hom_iso' (F : J ⥤ C) [has_colimit F] (W : C) :
   (colimit F ⟶ W) ≅ { p : Π j, F.obj j ⟶ W // ∀ {j j'} (f : j ⟶ j'), F.map f ≫ p j' = p j } :=
 (colimit.is_colimit F).hom_iso' W
 
+lemma colimit.desc_extend (F : J ⥤ C) [has_colimit F] (c : cocone F) {X : C} (f : c.X ⟶ X) :
+  colimit.desc F (c.extend f) = colimit.desc F c ≫ f :=
+begin
+  ext1, simp at *, erw ←category.assoc, simp, refl
```

Comment:
Not a fan of `simp, erw, simp`.

How about this: Add another lemma `category.assoc_symm` which is `category.assoc` with the two sides reversed. Then try to change this proof to `simp [-category.assoc, category.assoc_symm]`.

Reply:
Ok. This works even without the `-category.assoc`, so I've done that for now. I'm not sure why this doesn't cause looping, however!